packages/@react-spectrum/tag/src/Tag.tsx

```diff
@@ -10,82 +10,79 @@
  */
 
-import Alert from '@spectrum-icons/workflow/Alert';
-import {classNames, useStyleProps} from '@react-spectrum/utils';
+import {classNames, SlotProvider, useSlotProps, useStyleProps} from '@react-spectrum/utils';
 import {ClearButton} from '@react-spectrum/button';
-import {FocusRing} from '@react-aria/focus';
 import {mergeProps} from '@react-aria/utils';
-import React from 'react';
-import {SpectrumTagProps} from '@react-types/tag';
+import React, {useRef} from 'react';
 import styles from '@adobe/spectrum-css-temp/components/tags/vars.css';
+import {Text} from '@react-spectrum/text';
+import {useFocusRing} from '@react-aria/focus';
 import {useHover} from '@react-aria/interactions';
 import {useTag} from '@react-aria/tag';
-import {useTagGroupProvider} from './TagGroup';
 
-export const Tag = ((props: SpectrumTagProps) => {
+export function Tag(props) {
   const {
     isDisabled,
     isRemovable,
-    validationState,
+    item,
+    state,
+    onRemove,
     ...otherProps
   } = props;
   let {styleProps} = useStyleProps(otherProps);
   let {hoverProps, isHovered} = useHover({isDisabled});
-  const {
-    isDisabled: isGroupDisabled,
-    isRemovable: isGroupRemovable,
-    validationState: groupValidationState,
-    onRemove,
-    role
-  } =  useTagGroupProvider();
-
-  let removable = isGroupRemovable !== undefined ? isGroupRemovable : isRemovable;
-  let disabled = isGroupDisabled !== undefined ? isGroupDisabled : isDisabled;
-  let isInvalid = (validationState !== undefined ? validationState : groupValidationState) === 'invalid';
+  let {isFocusVisible, focusProps} = useFocusRing({within: true});
   let {clearButtonProps, labelProps, tagProps} = useTag({
     ...props,
-    isRemovable: removable,
-    isDisabled: disabled,
-    validationState: validationState !== undefined ? validationState : groupValidationState,
-    onRemove: props.onRemove || onRemove,
-    role
-  });
-  let {role: buttonRole, ...otherButtonProps} = clearButtonProps;
-  let icon = props.icon || (isInvalid && <Alert />);
+    isRemovable,
+    isDisabled,
+    item,
+    onRemove
+  }, state);
+
+  return (
+    <div
+      {...otherProps}
+      {...styleProps}
+      {...mergeProps(tagProps, hoverProps, focusProps, labelProps)}
+      role={tagProps.role}
+      className={classNames(
+        styles,
+        'spectrum-Tags-item',
+        {
+          'is-disabled': isDisabled,
+          'focus-ring': isFocusVisible,
+          'is-focused': isFocusVisible,
+          'spectrum-Tags-item--removable': isRemovable,
+          'is-hovered': isHovered
+        },
+        styleProps.className
+      )}>
+      <SlotProvider
+        slots={{
+          icon: {UNSAFE_className: classNames(styles, 'spectrum-Tag-icon'), size: 'XS'},
+          text: {UNSAFE_className: classNames(styles, 'spectrum-Tag-content')},
+          tagRemoveButton: {UNSAFE_className: classNames(styles, 'spectrum-Tag-action')}
```

Comment:
because you provide the tagRemoveButton, you don't technically need this slot, you can just pass the prop, `UNSAFE_className`, directly to TagRemoveButton